### ISSUE TITLE
Flows cannot explicitly request a transaction be notarized

### ISSUE THREAD

**Issue by siddtwm** [corda/corda#2913]
In corda-core (**release-V3.0** on commit  **d9d6684**), the FinalityFlow determines whether the signed transaction should be notarized based on presence of inputs, time window and by the presence of the notary's key in the list of signers. This approach does not allow the flow creation process to explicitly request notarization of the transaction. 

The presence of the notary's signature will be helpful if any of the participants on the newly generated contract state demand that every newly issued contract be signed off by the validator.

Running any flow that only creates a single contract state while the validator node is not running demonstrates this issue.

**Comment by mikehearn** [comment]
So to clarify you want to notarise a transaction that is purely an issuance, even though it doesn't have a timestamp?

The reason we don't allow this is that the notary wouldn't check or do anything. It'd just sign, but the signature wouldn't mean anything (unless it was a validating notary, in which case it'd have validated, but the recipient is supposed to do that anyway).

Can you explain a bit more about why you'd like to do this? Does it meet some business requirement we overlooked?

**Comment by joeldudleyr3** [comment]
@siddtwm See Mike's questions above :)

**Comment by siddtwm** [comment]
Yes, indeed. We only use the validating notary in our use-case. From our perspective issuance notarization has to be one of the functions validator provides - ensure that all new contract states created by participants meet our definition of state instance. This is, of course over and above the functions Corda's validating notary already has. Further, if we introduce special audit notaries, they too may have custom logic around what constitutes a valid contract state instance.

**Comment by mikehearn** [comment]
@siddtwm If a party issues a state, then it resides only in their own vault and thus nobody else cares. If they then send it to a counterparty then that tx will be validated by the notary and only signed if it's valid, so the validity of the move tx captures the validity of the issuance.

So I'm still not quite sure what notarisation of issuances without timestamps will accomplish.

**Comment by lankydan** [comment]
Since this has received no further feedback, would it be best to pose this as a question to the dev mailing list and see what others think? @mikehearn

**Comment by roger3cev** [comment]
Also don't understand the need for a signature over a transaction with no input states or timestamp. Given that the notary only checks for input state uniqueness and/or acts as a time stamping authority, if it signs a transaction with a absence of both, then it doesn't mean anything. 

In terms of validation, all participants do this anyway. Maybe I'm missing something?
